File: nomad-samples/js/content-admin-api/content/update-content-definition.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function updateContentDefinition(AUTH_TOKEN, CONTENT_DEFINITION_ID, CREATE_DATE, LAST_MODIFIED_DATE, EDITOR_TEMPLATE, SAMPLE_TEMPLATE, IS_SYSTEM_MODULE, USE_EDITOR_FORM_OVERRIDE, TEMPLATE_FOLDER_ASSET_ID, CONTENT_DEFINITION_GROUP, CONTENT_TYPE_GROUP) 
{
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
  	HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);

    const BODY = {};

    const ARGUMENTS = ["createDate", "lastModifiedDate", "editorTemplate", "sampleTemplate", "isSystemModule", "useEditorFormOverride", "templateFolderAssetId"]; 
    for (let argNum = 0; argNum < ARGUMENTS.length; ++argNum)
    {
        let argsNum = argNum + 2
        if (arguments[argsNum] != "")
        {
            BODY[ARGUMENTS[argNum]] = arguments[argsNum];
        }
    }

    BODY["properties"] = {};
    if (CONTENT_DEFINITION_GROUP != null)
    {
        BODY.properties["ContentDefinitionGroupId"] = CONTENT_DEFINITION_GROUP;
    }
    if (CONTENT_TYPE_GROUP != null)
    {
        BODY.properties["ContentTypeId"] = CONTENT_TYPE_GROUP;
    }

    // Post
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/contentDefinition/${CONTENT_DEFINITION_ID}`, {
        method: "PUT",
        headers: HEADERS,
        body: JSON.stringify(BODY),
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        // Get the JSON from the response
        const INFO = await RESPONSE.json();

        return INFO;
    }
		
  	// There was an error
    await apiExceptionHandler(RESPONSE, "Update content definition failed");
}

File: nomad-samples/python/content_admin_api/content/update_content_definition.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def update_content_definition(AUTH_TOKEN: str, ID: str) -> dict:

    # Check for valid parameters
    if (not AUTH_TOKEN):
        raise Exception("Authentication Token: The authentication token is invalid")

    API_URL = ADMIN_API_URL + "/contentDefinition/" + ID
        
    # Create header for the request
    HEADERS = {
      	"Authorization": "Bearer " + AUTH_TOKEN,
        "Content-Type": "application/json"
    }

    BODY = {
        "contentDefinitionId": "95638412-7a40-4710-8b0e-329191f18a74",
        "createdDate": "2021-08-03T18:30:50.53Z",
        "lastModifiedDate": "2022-09-17T11:13:03.028Z",
        "isSystemModule": False,
        "useEditorFormOverride": False,
        "templateFolderAssetId": "687c59bd-59b8-4026-9b7f-6f469e05cfdb"
    }

    try:
        # Send the request
        RESPONSE = requests.put(API_URL, headers= HEADERS, data=json.dumps(BODY))

        if not RESPONSE.ok:
            raise Exception("Response returned " + str(RESPONSE.status_code))


    except:
        api_exception_handler(RESPONSE, "Update Content Definition Failed")

